### PR TITLE
Adjust documentation to the new Docker Hub location and simplify the demo

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] Adjust documentation to the new Docker Hub location
- [x] Remove need in root user and JAVA_OPTS
- [x] Simplify the guidelines

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,6 +12,9 @@ USER root
 RUN mkdir -p /usr/share/jenkins/ && ln -s /usr/share/cloudbees-jenkins-distribution/cloudbees-jenkins-distribution.war /usr/share/jenkins/jenkins.war
 USER cloudbees-jenkins-distribution
 
+# Startup all plugins included into the CloudBees Jenkins Distribution bundle
+ENV JAVA_OPTS "-Dcom.cloudbees.jenkins.cjp.installmanager.CJPPluginManager.allRequired=${INCLUDE_ALL_BUNDLED_PLUGINS}"
```
Comment (fcojfernandez) [inline Dockerfile]: what if `INCLUDE_ALL_BUNDLED_PLUGINS` has not been set?

File: Makefile [hunk 2]
```diff
@@ -1,25 +1,21 @@
...
 run-devel:
-	@if ls work ; then rm -rf work ; false; fi
+	rm -rf work
 	mkdir work
-	docker run -u root --rm -p 8080:8080 -p 50000:50000 \
-		-v /var/run/docker.sock:/var/run/docker.sock \
-		-e JAVA_OPTS="-Dcom.cloudbees.jenkins.cjp.installmanager.CJPPluginManager.allRequired=${INCLUDE_ALL_BUNDLED_PLUGINS}" \
+	docker run --rm -p 8080:8080 \
 		-e VERBOSE=true \
 		-v $(CURDIR)/work:/var/jenkins_home/ \
```
Comment (fcojfernandez) [inline Makefile]: what about `/var/cloudbees-jenkins-distribution`?

File: Dockerfile (hunk 1 above)
Comment (oleg-nenashev) [reply Dockerfile]: Fixing it now

File: Makefile (hunk 2 above)
Comment (oleg-nenashev) [reply Makefile]: Did not need it. It is a developer code.
Is it a symlink?

Comment (fcojfernandez) [reply Makefile]: The Dockerfile is based on a CJD docker image whose JENKINS_HOME is `/var/cloudbees-jenkins-distribution` and not `/var/jenkins_home/` (`FROM cloudbees/cloudbees-jenkins-distribution:2.176.2.3`), so when you say
> It is a developer code.

Can I understand you are not interested in storing the JENKINS_HOME?

Comment (oleg-nenashev) [reply Makefile]:
```suggestion
		-v $(CURDIR)/work:/var/cloudbees-jenkins-distribution/ \
```

Right, just forgot to patch it after updating the baseline

**Review by fcojfernandez — APPROVED** [review]

**Pull request merged**